tests/jerry/array-prototype-splice.js

```diff
@@ -175,3 +175,228 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+// Checking behavior when the function's this_argument is undefined
+try {
+  Array.prototype.splice.call(undefined);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+// Checking behavior when length is an object, which throws error
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("foo"); } });
+  var a = { length : o };
+  Array.prototype.splice.call(a);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "foo");
+}
+
+// Checking behavior when the first argument of the function is an object, which throws error
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("1"); } });
+  [1, 2].splice(o);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "1");
+}
+
+// Checking behavior when the second argument of the function is an object, which throws error
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("2"); } });
+  [1, 2].splice(1, o);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "2");
+}
+
+// Checking behavior when the third element of the array throws error
+try {
+  var a = [1, 5, ,2];
+  Object.defineProperty(a, '2', { 'get' : function() { throw new ReferenceError("3"); } });
+  a.splice(1, 7, 7, 7, 7, 7);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "3");
+}
+
+// Checking behavior when the function's this_argument is an object, witch only contains a length
+var a = {length : 13};
+Array.prototype.splice.call(a, function () { delete a; });
+assert(a.length === 0);
+
+// Checking behavior when the first element of the array throws error
+try {
+  var a = [1, 5, 6, 7, 8, 5];
+  Object.defineProperty(a, '0', { 'get' : function() { throw new ReferenceError("foo0"); } });
+  Array.prototype.splice.call(a, 0, 3);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "foo0");
+}
+
+// Checking behavior when a modified object is an element of the array
+try {
+  f = function()
+      {
+        delete arr[3];
+        arr.length = 13;
+        Object.defineProperty(arr, '5', function() { });
+      };
```

Comment:
This style has became even worse than it was before. Please use the following format:
``` js
f = function() {
  delete arr[3];
  arr.length = 13;
  Object.defineProperty(arr, '5', function() { });
};
```